blog/some-challenges-of-being-the-editor-of-a-software-intended-to-be-integrated.mdx

```diff
@@ -0,0 +1,104 @@
+- a merchant willing to sell something to…
+- a customer seeking a product.
+
+Anybody having at least a small experience as a software editor knows that
+properly addressing one persona can already be challenging, so you can guess
+what it is like with three, even if they don’t necessarily want opposing
+features, more targets mean more constraints.
+
+Let’s take a deliberately extreme example 😃 They all want a performant store.
+
+The customer:
+
+> I want a performant and great shopping experience even in the middle of
+> nowhere with a low to average 3G connection.
+
+The merchant:
+
+> Can I have a great shopping experience for my customers? And also, I want to
+> put very high quality photos and videos online. In addition, I want my pages
+> to be always highly up to date with customer specific and very dynamic
+> content. And next Friday is Black Friday! The website needs to scale,
+> it’s the biggest selling event of the year.
+
+The developers:
+
+> The merchant wants a performant and scalable website. Is your software doing
+> that out of the box or can it be done by turning on a setting?
+
+I guess you get the idea, each target adds a set of constraints. It’s no
+surprise that pleasing  everyone is difficult but it’s also a stimulating
+challenge.
+
+## Extensibility
+
+As a software meant to be integrated by developers, Front-Commerce needs to be
+extensible so that it can serve different use cases. The most obvious reason is
+that each merchant wants its own customized store implementing a dedicated
+shopping experience.
+
+In addition, each project needs a specific set of features and while we try to
+support as many as possible out of the box, having an extensible architecture
+allows the developers to fill the gaps. As far as I’m concerned, it’s always a
+pleasure to see very unexpected or specific features nicely implemented even if
+I would never have imagined it could exist.
+
+It’s also worth mentioning that in the world of composable architectures and
+[headless commerce](https://www.front-commerce.com/what-is-headless-commerce/),
+extensibility is a key feature.
+
+## Backwards compatibility and *upgradability*
+
+Being able to implement a project is nice but that’s only the very beginning of
+the story. At some point, new features and fixes for bugs will be needed. The
+challenge here is to be able to deliver those without breaking anything.
+
+On a practical level, that means we try to follow [semantic
+versioning](https://semver.org/) by regularly providing patch releases
+containing only bug fixes and every six weeks, we release a minor version with
+new features. Those minor versions always have [a migration
+guide](https://developers.front-commerce.com/docs/appendices/migration-guides/)
+to make the upgrade process as frictionless as possible. And what about a major
+version? In the last couple of years, we have not released any, but I’ve heard
+that something is cooking 😇
```

Comment:
**suggestion:** I think it's better to mention the overall vision of what is needed instead.

Here is a try. The last sentence is optional, I thought it could make sense but feel free to ignore it if needed!
```suggestion
version? In the last couple of years, we have not released any, but it couldn't be done without a solid migration automation (codemods) to ensure no-one is left behind! We've heard about major versions that required weeks of basically rewriting everything… this isn't viable for most projects and leads to outdated software running in production years after.
```

Reply:
well, the initial idea was to tease a bit and to be concise, the emoji is also to give a bit a rythm to the text.

But it's an interesting aspect, maybe we can keep it for a post dedicated to the
v3 ?